diff --git a/TPMCmd/tpm/include/BaseTypes.h b/TPMCmd/tpm/include/BaseTypes.h
index afcfef9..bf589f0 100644
--- a/TPMCmd/tpm/include/BaseTypes.h
+++ b/TPMCmd/tpm/include/BaseTypes.h
@@ -40,11 +40,6 @@
 #ifndef _BASE_TYPES_H_
 #define _BASE_TYPES_H_
 
-// NULL definition
-#ifndef NULL
-#define NULL                (0)
-#endif
-
 typedef uint8_t             UINT8;
 typedef uint8_t             BYTE;
 typedef int8_t              INT8;
